### PR TITLE
solution added

### PR DESCRIPTION
**Pull request by IrynaDim**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -27,4 +27,33 @@ public Integer getLength() {
...
+        if (this == object) {
+            return true;
+        }
+        if (object == null) {
+            return false;
+        }
+        if (object.getClass().equals(Rectangle.class)) {
+            Rectangle rectangle = (Rectangle) object;
+            return (color != null && color.equals(rectangle.color))
+                    || (color == null && rectangle.color == null)
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: can we simplify it?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +27,33 @@ public Integer getLength() {
...
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 14;
+        result = 28 * result + (color == null ? 0 : color.hashCode());
+        result = 28 * result + (width == null ? 0 : width);
+        result = 28 * result + (length == null ? 0 : length);
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Rectangle.java]: why 14 and 28?

Comment (IrynaDim) [reply src/main/java/core/basesyntax/Rectangle.java]: I dont know, its random numbers. Bohdan in the video choose 17 and 31. I was shure that i can choose whatever i want either.

Comment (IrynaDim) [reply src/main/java/core/basesyntax/Rectangle.java]: i have added 31 and 17

Comment (dmytro-hryhoruk) [reply src/main/java/core/basesyntax/Rectangle.java]: You can choose any number but it should be a primary one and 31 is the most popular

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request closed without merging**